### PR TITLE
09485 9448 pcli wiring diagram

### LINKED ISSUE #9485: Create PCLI utility for generating mermaid wiring diagram
Create a PCLI utility that generates a mermaid style wiring diagram from `PlatformWiring.java`.

### PR DESCRIPTION
**Pull request by cody-littley**
Closes #9485

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/cli/DiagramCommand.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+    private List<String> groupStrings = List.of();
+    private List<String> collapsedGroupStrings = List.of();
+
+    private DiagramCommand() {}
+
+    @CommandLine.Option(
+            names = {"-g", "--group"},
+            description =
+                    "Specify a an un-collapsed grouping. " + "Format is 'GROUP_NAME:COMPONENT_NAME[,COMPONENT_NAME]*'.")
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/cli/DiagramCommand.java:54]: Specify ~~a~~ an...

Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/cli/DiagramCommand.java:54]: unnecessary concatenation

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/cli/DiagramCommand.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    private void setGroupStrings(@NonNull final List<String> groupStrings) {
+        this.groupStrings = groupStrings;
+    }
+
+    @CommandLine.Option(
+            names = {"-c", "--collapsed-group"},
+            description = "Specify a collapsed grouping. " + "Format is 'GROUP_NAME:COMPONENT_NAME[,COMPONENT_NAME]*'.")
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/cli/DiagramCommand.java:61]: Unnecessary concatenation

**Review by alittley — APPROVED** [review]

**Review by lpetrovic05 — DISMISSED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/cli/DiagramCommand.java (hunk 1 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/cli/DiagramCommand.java:54]: fixed

Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/cli/DiagramCommand.java:54]: dang it spotless! Fixed.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/cli/DiagramCommand.java (hunk 2 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/cli/DiagramCommand.java:61]: fixed

**Review by alittley — APPROVED** [review]

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/wires/input/BindableInputWire.java [hunk 3]
```diff
@@ -76,6 +76,7 @@ public void bind(@NonNull final Function<IN, OUT> handler) {
                 taskSchedulerInput.forward(output);
             }
         });
+        model.registerInputWireBinding(taskSchedulerName, getName());
```
Comment (alittley) [inline platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/wires/input/BindableInputWire.java:79]: This is being called twice now

Comment (cody-littley) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/wires/input/BindableInputWire.java:79]: Doh! Fixed.

**Review by alittley — APPROVED** [review]

**Pull request merged**